pkg/dns/dns.go

```diff
@@ -48,28 +49,38 @@ const (
 
 	// Resync period for the kube controller loop.
 	resyncPeriod = 30 * time.Minute
+
+	// Duration for which the TTL cache should hold the node.
+	nodeCacheTTL = 180 * time.Second
 )
 
 type KubeDNS struct {
 	kubeClient *kclient.Client
 	// DNS domain name.
 	domain string
+	// Map of federation names that the cluster in which this kube-dns is running belongs to, to
+	// the corresponding domain names.
+	federations map[string]string
 	// A cache that contains all the endpoints in the system.
 	endpointsStore kcache.Store
 	// A cache that contains all the services in the system.
-	servicesStore     kcache.Store
+	servicesStore kcache.Store
+	// A TTL cache that contains some subset of nodes in the system.
+	nodesStore        kcache.Store
```

Comment:
Added a comment to clarify this a bit. Please let me know if that is sufficient.